source/Engine.cpp

```diff
@@ -868,12 +864,34 @@ void Engine::Step(bool isActive)
 	}
 
 	// Draw crosshairs on any minables in range of the flagship's scanners.
-	double scanRange = flagship ? 100. * sqrt(flagship->Attributes().Get("asteroid scan power")) : 0.;
+	// Do not use the sqrt here or for the length.
+	double scanRange = flagship ? 10000. * flagship->Attributes().Get("asteroid scan power") : 0.;
 	if(flagship && scanRange && !flagship->IsHyperspacing())
+	{
+		// Decide before looping whether or not to catalog asteroids.  This
+		// results in cataloging in-range asteroids roughly 3 times a second.
+		bool shouldCatalogAsteroids = (!isAsteroidCatalogComplete && !Random::Int(20));
+		bool scanComplete = true;
 		for(const shared_ptr<Minable> &minable : asteroids.Minables())
 		{
+			// In this case, center is the flagship's position.
 			Point offset = minable->Position() - center;
-			if(offset.Length() > scanRange && flagship->GetTargetAsteroid() != minable)
+			// Use the squared length, as we used the squared scan range.
+			bool inRange = offset.LengthSquared() <= scanRange;
+
+			// Autocatalog asteroid: Record that the player knows this type of asteroid is available here.
+			if(shouldCatalogAsteroids && !asteroidsScanned.count(minable->Name()))
+			{
+				scanComplete = false;
+				if(!Random::Int(10) && inRange)
+				{
+					asteroidsScanned.insert(minable->Name());
```

Comment:
Why do you use the name, instead of the pointer?

Reply:
Open to suggestion; I wasn't sure how to track it just that it needs to be tracked to disable catalog when it is complete.

Reply:
If there's two 'Gold' asteroids in the system will the be guaranteed to have the same pointer?  If so we could switch to pointer.  If not, then the name is necessary to know when scanning needs to stop for the current system.

Scanning would stop when all asteroids have been logged in the ship computer.

Reply:
They would have different pointer, since they would be different objects.